compiler/rustc_macros/src/lib.rs

```diff
@@ -24,6 +26,27 @@ pub fn symbols(input: TokenStream) -> TokenStream {
     symbols::symbols(input.into()).into()
 }
 
+/// Creates a struct type `S` that can be used as an index with
+/// `IndexVec` and so on.
+///
+/// There are two ways of interacting with these indices:
+///
+/// - The `From` impls are the preferred way. So you can do
+///   `S::from(v)` with a `usize` or `u32`. And you can convert back
+///   to an integer with `u32::from(s)`.
```

Comment:
I know this comment is pre-existing, but while we're here: is this advice correct? I personally prefer `S::new(v)` and `s.index()`, and I feel like I've seen those forms more often.

Reply:
I'm not sure - I think this could be dealt with in a follow-up issue, since it's just a style reccomendation.